deployer/deployer.py

```diff
@@ -297,23 +295,61 @@ def deploy_version_without_eureka(self, version, health_check, health_check_port
             }
         }
 
-        request = self.request("deployment/start", json.dumps(data))
-
-        time.sleep(30)
+        success = self.start_deployment(data, version)
+        if success:
+            self.resize_asg(asg_name=version)
 
-        self.resize_asg(asg_name=version)
+            good_deploy = self.wait_asg_ready(version=version,
+                                              health_check=health_check,
+                                              health_check_port=health_check_port)
 
-        good_deploy = self.wait_asg_ready(version=version,
-                                          health_check=health_check,
-                                          health_check_port=health_check_port)
+            if good_deploy:
+                if remove_old:
+                    self.remove_old_asg(new_asg_name=version)
+            else:
+                self.disable_asg(asg_name=version)
 
-        if good_deploy:
-            if remove_old:
-                self.remove_old_asg(new_asg_name=version)
         else:
-            self.disable_asg(asg_name=version)
+            raise Exception("Fucking asgard!")
+
+    def start_deployment(self, data, version):
+        wait_seconds_after_stack_start = 10
+        retries = 0
+        success = False
+        while not success and retries < 10:
+            result = self.request("deployment/start", json.dumps(data))
+            id_task = json.loads(result.content)["deploymentId"]
+            time.sleep(wait_seconds_after_stack_start)
+            success = self.check_task_creation(id_task)
+            if not success:
+                retries += 1
+                auto_scaling_check = self.check_for_auto_scaling_group_creation(version)
+                if auto_scaling_check:
+                    raise Exception("Error: Autoscaling is up, but instances not working propertly")
+                else:
+                    print("Autoscaling error")
+        return success
+
+    def check_task_creation(self, id_task):
```

Comment:
@Arcontus the same of the other function, this function doesn't really check the task creation, this method wait until the task finish... change the name please.